### PR TITLE
Gizmo snapping

### PR DESCRIPTION
**Pull request by TrevorDev**
https://github.com/BabylonJS/Babylon.js/issues/4141

### CODE REVIEW

File: src/Gizmos/babylon.axisDragGizmo.ts [hunk 1]
```diff
@@ -5,6 +5,15 @@ module BABYLON {
...
         private _pointerObserver:Nullable<Observer<PointerInfo>> = null;
+        /**
+         * Drag distance in babylon units that the gizmo will snap to when dragged (Default: 0)
+         */
+        public snapDistance = 0;
+        /**
+         * Event that fires each time the gizmo snaps to a new location.
+         * * snapDistance is the the change in distance
+         */
+        public onSnapObservable = new Observable<{snapDistance:number}>();
```
Comment (sebavan) [inline src/Gizmos/babylon.axisDragGizmo.ts]: i am not seeing the clear in dispose but on a mobile :-)

File: src/Gizmos/babylon.axisDragGizmo.ts [hunk 2]
```diff
@@ -42,16 +51,28 @@ module BABYLON {
...
+                        this.attachedMesh.position.addInPlace(event.delta);
+                    }else{
+                        currentSnapDragDistance+=event.dragDistance
+                        if(Math.abs(currentSnapDragDistance)>this.snapDistance){
+                            var dragSteps = Math.floor(Math.abs(currentSnapDragDistance)/this.snapDistance);
+                            currentSnapDragDistance = currentSnapDragDistance % this.snapDistance;
+                            event.delta.normalizeToRef(tmpVector);
+                            tmpVector.scaleInPlace(this.snapDistance*dragSteps);
+                            this.attachedMesh.position.addInPlace(tmpVector);
+                            this.onSnapObservable.notifyObservers({snapDistance: this.snapDistance*dragSteps});
```
Comment (sebavan) [inline src/Gizmos/babylon.axisDragGizmo.ts]: should we keep a tmp object for this ?

File: src/Gizmos/babylon.axisScaleGizmo.ts [hunk 3]
```diff
@@ -5,6 +5,15 @@ module BABYLON {
     export class AxisScaleGizmo extends Gizmo {
         private _dragBehavior:PointerDragBehavior;
         private _pointerObserver:Nullable<Observer<PointerInfo>> = null;
+        /**
```
Comment (sebavan) [inline src/Gizmos/babylon.axisScaleGizmo.ts]: could the code be factorized with the previous one ? it sounds similar with the same dispose comment

File: src/Gizmos/babylon.axisDragGizmo.ts (hunk 1 above)
Comment (TrevorDev) [reply src/Gizmos/babylon.axisDragGizmo.ts]: added clear

**Pull request merged**